src/astarte-client/types/dataType.ts

```diff
@@ -16,4 +16,42 @@
    limitations under the License.
 */
 
-export type AstarteDataType = number | boolean | string | number[] | boolean[] | string[];
+export type AstarteDataValue = number | boolean | string | number[] | boolean[] | string[] | null;
+
+export type AstarteDataType =
+  | 'double'
+  | 'integer'
+  | 'boolean'
+  | 'longinteger'
+  | 'string'
+  | 'binaryblob'
+  | 'datetime'
+  | 'doublearray'
+  | 'integerarray'
+  | 'booleanarray'
+  | 'longintegerarray'
+  | 'stringarray'
+  | 'binaryblobarray'
+  | 'datetimearray';
+
+export type AstarteDataTuple =
```

Comment:
I would call this AstarteValue, because it represents a single value sent (or received) by the device.
Note that a single value can also be represented with an array, like the type `integerarray`